File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
answer = 0

def dfs(n, num,  numbers, target):
    global answer
    if n == len(numbers):
        if num == target:
            answer += 1
        return
    dfs(n+1, num+numbers[n], numbers, target)
    dfs(n+1, num-numbers[n], numbers, target)


def solution(numbers, target):
    dfs(0, 0,  numbers, target)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    const len = numbers.length;
    
    function dfs(idx, sum) {
        if(idx === len) {
            if(sum === target) answer++;
            return;
        }
        
        dfs(idx+1, sum+numbers[idx]);
        dfs(idx+1, sum-numbers[idx]);
    }
    
    dfs(0,0);
    
    return answer;
}